### PR TITLE
fix(makefile): fix gopls error, makefile help

### PR DESCRIPTION
**Pull request by hookenz**
Some improvements to the Makefile and also removing a `gopls` error due to the go.mod in `docker-credential-portainer` which has one at the top level.

Added `go.work` file

Closes: [DTD-121](https://portainer.atlassian.net/browse/DTD-121)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -12,22 +12,24 @@ agent=agent
...
 
-download-binaries:
+download-binaries: ## Download dependant binaries
 	@./setup.sh $(PLATFORM) $(ARCH)
 
-clean:
+clean: ## Remove all build and download artifacts
 	@rm -f dist/*
 
+help:
```
Comment (chiptus) [inline Makefile]: add this as a default target?

File: cmd/docker-credential-portainer/go.mod [hunk 2]
```diff
@@ -1,5 +0,0 @@
-module github.com/portainer/docker-credential-portainer
```
Comment (chiptus) [inline cmd/docker-credential-portainer/go.mod:1]: as discussed, I think it's better to have this as a separate module and use `go.work`

File: Makefile (hunk 1 above)
Comment (hookenz) [reply Makefile]: I'd like to, need to check CI process

File: cmd/docker-credential-portainer/go.mod (hunk 2 above)
Comment (hookenz) [reply cmd/docker-credential-portainer/go.mod:1]: But it's still a cmd so follows the recommended practice of having commands inside cmd directory.
By moving it to the top level, we're breaking that practise and it'll make navigating sources harder if we add more binaries in other directories at the top level.   The way I've fixed it seems right IMO.

Also I think go.work is for modules you're importing to your project not the other way around.  It allows go sub-commands to operate on multiple modules at once.  So you do not need to add a hack to go.mod with replace anymore.

Comment (hookenz) [reply cmd/docker-credential-portainer/go.mod:1]: Ok, it seems to work.  I think we should use `go.work` but keep the command in the sources beneath the `cmd` directory.

**Review by dakota-portainer — APPROVED** [review]
Simple and clean. I like comments you added in the makefile.

**Pull request merged**